src/System.Private.CoreLib/shared/System/Text/Utf16Utility.cs

```diff
@@ -115,15 +124,92 @@ internal static bool UInt32ContainsAnyUppercaseAsciiChar(uint value)
             Debug.Assert(AllCharsInUInt32AreAscii(value));
 
             // the 0x80 bit of each word of 'lowerIndicator' will be set iff the word has value >= 'A'
-            uint lowerIndicator = value + 0x00800080u - 0x00410041u;
+            uint lowerIndicator = value + 0x0080_0080u - 0x0041_0041u;
 
             // the 0x80 bit of each word of 'upperIndicator' will be set iff the word has value > 'Z'
-            uint upperIndicator = value + 0x00800080u - 0x005B005Bu;
+            uint upperIndicator = value + 0x0080_0080u - 0x005B_005Bu;
 
             // the 0x80 bit of each word of 'combinedIndicator' will be set iff the word has value >= 'A' and <= 'Z'
             uint combinedIndicator = (lowerIndicator ^ upperIndicator);
 
-            return (combinedIndicator & 0x00800080u) != 0;
+            return (combinedIndicator & 0x0080_0080u) != 0;
+        }
+
+        /// <summary>
+        /// Given two UInt32s that represent two ASCII UTF-16 characters each, returns true iff
+        /// the two inputs are equal using an ordinal case-insensitive comparison.
+        /// </summary>
+        /// <remarks>
+        /// This is a branchless implementation.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool UInt32OrdinalIgnoreCaseAscii(uint valueA, uint valueB)
```

Comment:
Method name should have `Equals` in it as it doesn't specify what its doing currently